### PR TITLE
[14.0][IMP] account_payment_term_discount (Allow bulk payment if no discount)

### PR DESCRIPTION
**Pull request by patrickrwilson**
During bulk or grouped payments, this checks if there are any invoices with a possible discount and if exist, it stops the payment process. Otherwise the process will continue allowing for bulk or grouped payments. 

This is a simple solution that addresses issue #414

### CODE REVIEW

**Comment by patrickrwilson** [comment]
@dreispt Here is the simple solution to issue #414 for your review.

**Comment by dreispt** [comment]
Hello @patrickrwilson , I just learned there is WIP to implement this support at #398, that has a dependency on #391, also undergoing fixes.
@murtuzasaleh is working on it.

**Comment by patrickrwilson** [comment]
> Hello @patrickrwilson , I just learned there is WIP to implement this support at #398, that has a dependency on #391, also undergoing fixes.
> @murtuzasaleh is working on it.

@dreispt Yes, they are related however I think this simple solution still would be useful in the event a person only uses this module and not the others that are currently being worked. That way they would still be able to use the out of box batch payments. Correct me if there are other thoughts.

File: account_payment_term_discount/models/account_payment.py [hunk 1]
```diff
@@ -90,14 +90,17 @@ def onchange_payment_amount(self):
...
         if any(len(active_id) != 1 for record in self):
-            # For multiple invoices, there is account.register.payments wizard
-            raise UserError(
-                _(
-                    "This method should only be called to process a "
-                    "single invoice's payment."
-                )
-            )
+            for invoice in self.line_ids:
+                if invoice.move_id.discount_amt > 0:
```
Comment (dreispt) [inline account_payment_term_discount/models/account_payment.py]:
```suggestion
            moves = self.mapped("lines_ids.move_id").filtered("discount_amt")
            if moves:
```

File: account_payment_term_discount/models/account_payment.py [hunk 2]
```diff
@@ -90,14 +90,17 @@ def onchange_payment_amount(self):
     def action_create_payments(self):
         active_id = self.env.context.get("active_ids", [])
 
+        # Don't allow bulk or grouped payments if an invoice has a possible discount.
         if any(len(active_id) != 1 for record in self):
```
Comment (dreispt) [inline account_payment_term_discount/models/account_payment.py]:
```suggestion
        if len(active_id) > 1:
```

**Comment by patrickrwilson** [comment]
@dreispt changes made, travis is red due to account_payment_return module, no errors on this module that I can see.

**Comment by RLeeGML** [comment]
@patrickrwilson  The issues with account_payment_return should be resolved, if you rebase it may fix travis check

**Comment by patrickrwilson** [comment]
@RLeeOSI Something is strange with my branch and it won't let me resolve conflicts because I don't have write access to my own branch, I'm going to close this and create a new one under a new branch. Sorry for the noice.

**Comment by patrickrwilson** [comment]
@RLeeOSI and after looking deeper, this issue looks to have been resolved with https://github.com/OCA/account-payment/pull/398 anyway so this PR is not needed anymore. I won't be opening a new one.

**Pull request closed without merging**